cli/roger_build.py

```diff
@@ -47,104 +53,122 @@ def parse_args(self):
+                if not file_exists:
+                    raise ValueError("Specified build file: {} does not exist. Exiting build.".format(build_filename))
+            else:
+                file_exists = os.path.exists("{0}/Dockerfile".format(file_path))
+
+            if file_exists:
+                if 'registry' not in roger_env:
+                    raise ValueError('Registry not found in roger-env.json file.')
+                image = "{0}/{1}".format(roger_env['registry'], args.tag_name)
+                try:
+                    if abs_path == args.directory:
+                        try:
+                            dockerObj.docker_build(
+                                dockerUtilsObj, appObj, args.directory, repo, projects, docker_path, image, build_filename)
+                        except ValueError:
+                            print('Docker build failed.')
+                            raise
+                    else:
+                        directory = '{0}/{1}'.format(cur_dir, args.directory)
+                        try:
+                            dockerObj.docker_build(
+                                dockerUtilsObj, appObj, directory, repo, projects, docker_path, image, build_filename)
+                        except ValueError:
+                            print('Docker build failed.')
+                            raise
+                    build_message = "Image {0} built".format(image)
+                    if(args.push):
+                        exit_code = dockerUtilsObj.docker_push(image)
+                        if exit_code != 0:
+                            raise ValueError(
+                                'Docker push failed.')
+                        build_message += " and pushed to registry {}".format(roger_env[
+                                                                             'registry'])
+                    print(build_message)
+                except (IOError) as e:
+                    print("The folowing error occurred.(Error: %s).\n" %
+                          e, file=sys.stderr)
+                    raise
+            else:
+                print("Dockerfile does not exist in dir: {}".format(file_path))
+
+            hookname = "post_build"
+            exit_code = hooksObj.run_hook(hookname, data, file_path)
+            if exit_code != 0:
+                raise ValueError('{} hook failed.'.format(hookname))
+        except (Exception) as e:
+            execution_result = 'FAILURE'
+            print("The following error occurred: %s" %
+                  e, file=sys.stderr)
+            raise
+        finally:
+            time_taken = time() - function_execution_start_time
+            input_metric = str(args.app_name) + ".roger_build," + "outcome=" + str(execution_result) + ",config_name=" + str(config_name) + ",environment=" + str(args.env) + ",user=" + str(settingObj.getUser()) + ",time_taken=" + str(time_taken) + " secs"
```

Comment:
1) Yes according to this code, it will end up as a "time_taken" tag. In order to send it as a timing metric data type, you need to send it as given here: https://influxdata.com/blog/getting-started-with-sending-statsd-metrics-to-telegraf-influxdb/

Example: 
response.time:301|ms
response.time:301|h

2) Why are you using the increment method of statsd? This is not a counter. Please use timing_ms(). 
Reference: https://metacpan.org/pod/Net::Statsd::Client#statsd-set_add-metric-value-sample_rate